mlir/lib/Conversion/MemRefToSPIRV/MemRefToSPIRV.cpp

```diff
@@ -481,6 +481,14 @@ calculateRequiredAlignment(Value accessedPtr, Operation *memrefAccessOp) {
   assert((isa<memref::LoadOp, memref::StoreOp>(memrefAccessOp)) &&
          "Bad op type");
 
+  auto nontemporalAttr = memrefAccessOp->getAttrOfType<BoolAttr>("nontemporal");
+  if (nontemporalAttr && nontemporalAttr.getValue()) {
+    return std::pair{
```

Comment:
Note that the `Nontemporal` is a bit in the bitfield--it should be or-ed together with the `Aligned` bit calculuated in the below--as it's implemented right now, this breaks the logic for alignment calculation.

We need to have a test on this (nontemporal + aligned) to check it.

Reply:
I updated the code to set `Nontemporal` only if `Aligned` is not set and added a test for this.